includes/class-amp-travel-taxonomies.php

```diff
@@ -0,0 +1,92 @@
+				'edit_item'                  => __( 'Edit Activity', 'travel' ),
+				'view_item'                  => __( 'View Activity', 'travel' ),
+				'update_item'                => __( 'Update Activity', 'travel' ),
+				'add_new_item'               => __( 'Add New Activity', 'travel' ),
+				'new_item_name'              => __( 'New Activity Name', 'travel' ),
+				'separate_items_with_commas' => __( 'Separate activities with commas', 'travel' ),
+				'add_or_remove_items'        => __( 'Add or remove activities', 'travel' ),
+				'choose_from_most_used'      => __( 'Choose from the most used activities', 'travel' ),
+				'not_found'                  => __( 'No activities found.', 'travel' ),
+				'no_terms'                   => __( 'No activities', 'travel' ),
+				'items_list_navigation'      => __( 'Activities list navigation', 'travel' ),
+				'items_list'                 => __( 'Activities list', 'travel' ),
+				'most_used'                  => __( 'Most Used', 'travel' ),
+				'back_to_items'              => __( '&larr; Back to Activities', 'travel' ),
+			),
+		) );
+
+		register_taxonomy( 'location', array( 'adventure', 'post' ), array(
+			'hierarchical'          => false,
+			'query_var'             => 'location',
+			'public'                => true,
+			'show_ui'               => true,
+			'show_admin_column'     => true,
+			'show_in_rest'          => true,
+			'rest_base'             => 'locations',
+			'rest_controller_class' => 'WP_REST_Terms_Controller',
+			'labels'                => array(
+				'name'                       => __( 'Locations', 'travel' ),
+				'singular_name'              => __( 'Location', 'travel' ),
+				'search_items'               => __( 'Search Locations', 'travel' ),
+				'popular_items'              => __( 'Popular Locations', 'travel' ),
+				'all_items'                  => __( 'All Locations', 'travel' ),
+				'edit_item'                  => __( 'Edit Location', 'travel' ),
+				'view_item'                  => __( 'View Location', 'travel' ),
+				'update_item'                => __( 'Update Location', 'travel' ),
+				'add_new_item'               => __( 'Add New Location', 'travel' ),
+				'new_item_name'              => __( 'New Location Name', 'travel' ),
+				'separate_items_with_commas' => __( 'Separate locations with commas', 'travel' ),
+				'add_or_remove_items'        => __( 'Add or remove locations', 'travel' ),
+				'choose_from_most_used'      => __( 'Choose from the most used locations', 'travel' ),
+				'not_found'                  => __( 'No locations found.', 'travel' ),
+				'no_terms'                   => __( 'No locations', 'travel' ),
+				'items_list_navigation'      => __( 'Locations list navigation', 'travel' ),
+				'items_list'                 => __( 'Locations list', 'travel' ),
+				'most_used'                  => __( 'Most Used', 'travel' ),
+				'back_to_items'              => __( '&larr; Back to Locations', 'travel' ),
+			),
+		) );
+	}
+}
+
+new AMP_Travel_Taxonomies();
```

Comment:
Note that in this case it would instead then be:

```
$amp_travel_taxonomies = new AMP_Travel_Taxonomies();
$amp_travel_taxonomies->init();
```

Unless the class switches to using static methods.

Reply:
Note also that `new AMP_Travel_Taxonomies()` shouldn't be in this file. The PHP file should just contain the one class. So `functions.php` should require the file and then do the above.

All of this to say, we may be complicating things 😄 Feel free to just add a new global function called `amp_travel_register_taxonomies()` and rename this file to `taxonomies.php` and include the `add_action( 'init', 'amp_travel_register_taxonomies' );` in the file. Since this is a theme you can feel free to have more “functional programming” instead of OOP.

Reply:
Thank you for the comments, @kienstra and @westonruter.

Based on @ThierryA's [comment](https://github.com/xwp/travel/pull/31#issuecomment-378860791) and @kienstra's [PR review on the blocks base layout](https://github.com/xwp/travel/pull/13#pullrequestreview-110950325) it seemed like using classes would be the preferred way. Happy to change it back to using global functions, however, maybe it would make sense to use the same structure throughout the code? For example if taxonomy-related functionality is not in OOP, should [Gutenberg block-related logic](https://github.com/xwp/travel/blob/feature/12-gutenberg_blocks_base/includes/class-amp-travel-blocks.php) still be in OOP? Or should we either use one or the other?

Thoughts?

Reply:
@miina I merged this into the PSTT theme which should give us a few structures. See https://github.com/xwp/travel/pull/38/files#diff-8ab8775c3cc79aadce48cad633a7fde3R109 
I will be looking at pulling in the structures for the blocks as well.

Reply:
I'm merging this now to unblock work, but this can remain an open conversation about whether to use OOP or global functions.